cpp/oneapi/dal/backend/memory.hpp

```diff
@@ -16,16 +16,18 @@
 
 #pragma once
 
-#ifdef ONEDAL_DATA_PARALLEL
-#include <CL/sycl.hpp>
-#endif
-
-#include "oneapi/dal/detail/common.hpp"
+#include "oneapi/dal/backend/common.hpp"
 
 namespace oneapi::dal::backend {
 
 #ifdef ONEDAL_DATA_PARALLEL
 
+inline bool is_device_accessible_usm_pointer(const sycl::queue& queue, const void* pointer) {
+    const auto pointer_type = sycl::get_pointer_type(pointer, queue.get_context());
+    return (pointer_type == sycl::usm::alloc::device) || //
+           (pointer_type == sycl::usm::alloc::shared);
```

Comment:
actually, every USM pointer except `sycl::usm::alloc::unknown` is device accessible

Reply:
I see, it makes sense to rename function. It's expected to return `true` if access to pointer is efficient on device. It's efficient in case of device USM and might be efficient in case of shared USM.